src/jet_server.cpp

```diff
@@ -29,65 +30,101 @@ void JetServer::addJetState(const std::string& path)
         std::string globalId = value.get("Global ID", "").asString();
         ComponentPtr component = componentIdDict.get(globalId);
         auto properties = component.getAllProperties();
-        try
+        for(auto property : properties)
         {
-            for(auto property : properties)
+            std::string propertyName = property.getName();
+            if (!value.isMember(propertyName))
+            {
+                std::cout << "addJetState: skipping property " << propertyName << std::endl;
+                continue;
+            }
+
+            Json::ValueType jsonValueType = value.get(propertyName, "").type();
+            bool typesAreCompatible = checkTypeCompatibility(jsonValueType, property.getValueType());
+            if(!typesAreCompatible)
+            {
+                throwJetModuleException(JetModuleException::JM_INCOMPATIBLE_TYPES);
+                continue;
+            }
+
+            std::cout << "addJetState: Changing value for property " << propertyName << ", type: " << jsonValueType << std::endl;
+            try
             {
-                std::string propertyName = property.getName();
-                Json::ValueType valueType = value.get(propertyName, "").type();
-
-                int64_t propertyValueInt;
-                uint64_t propertyValueUint;
-                double propertyValueDouble;
-                std::string propertyValueString;
-                bool propertyValueBool;
-                switch(valueType)
+                switch(jsonValueType)
                 {
-                    case Json::ValueType::nullValue:
-                        std::cout << "Null argument type detected" << std::endl;
-                        break;
                     case Json::ValueType::intValue:
-                        propertyValueInt = value.get(propertyName, "").asInt();
-                        component.setPropertyValue(propertyName, propertyValueInt);
+                        {
+                            int64_t oldValue = component.getPropertyValue(propertyName);
+                            int64_t newValue = value.get(propertyName, "").asInt64(); 
+                            if (oldValue != newValue)
+                                component.setPropertyValue(propertyName, newValue);
+                            else std::cout << "Value for " << propertyName << " has not changed. Skipping.." << std::endl;
```

Comment:
Please do a line break. This applies to all else. Our coding Guidelines says 
else
    std::cout ...